pkg/operator/controller/canary/controller.go

```diff
@@ -306,17 +311,45 @@ func (r *reconciler) startCanaryRoutePolling(stop <-chan struct{}) error {
 	return nil
 }
 
-func (r *reconciler) setCanaryFailingStatusCondition() error {
+func (r *reconciler) setCanaryFailingStatusCondition(errors []error) error {
+	errorStrings := deduplicateErrorStrings(errors)
+	if len(errorStrings) > 3 {
```

Comment:
Why 3 errors? I thought we wanted to show up to 5 deduplicated error messages. At least the PR description says so.

Reply:
Initially I had done 5 (since it takes 5 failures for canary to go degraded anyway), but I think it's too much to put in the status condition. I think 3 messages is more manageable, and still gives you some indication of if it's one particular error or multiple that's causing issues.

That said, I will update the description to match the current intended behavior.

Reply:
What do you think about putting this into a constant (similar to `canaryCheckFailureCount`)?

Reply:
I think that's a good idea. I will include a constant for it in my next pass.

Reply:
Did you mean to do this in your last update? or still working on it?

Reply:
I meant to do this in the last update, but looks like I missed it. I pushed a change that should fix it just a moment ago.